cmd/main.go

```diff
@@ -138,9 +141,17 @@ func main() {
 		os.Exit(1)
 	}
 
+	// creates the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		setupLog.Error(err, "unable to create clientset")
+		os.Exit(1)
+	}
```

Comment:
I forgot to mention why I did it that way: installing that package breaks the build. It triggers a series of updates and one of them is `api` at 0.31.1 which seems to have a lot of issues.

One thing I haven't checked is whether I would have the same issue on Linux.
Anyway, I thought it would be faster to have that implementation for now so it would be available and I would go back to check that build issue later on.

Reply:
@sgaist you can just install a specific version of the metrics client. We have api 0.28.3 for k8s so if you pick the same version you will not get a massive update.

```
go get k8s.io/metrics@v0.28.3
```

I just tried this quickly and I can definitely confirm that the pacakge was installed without updating anything else. Also I can confirm that it has the same functionality we need just as 0.31.1.

Reply:
Fixed using v0.28.3 package